Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs

```diff
@@ -40,26 +42,114 @@ public void ExecuteArgs(string[] args)
+            if (args[0] == "run" || args[0] == "grid")
             {
-                string[] arg1 = args[0].Split(new[]{ '='}, 2);
-                param = arg1[0].Trim();
-                value = arg1[1].Trim();                
+                ParseArgs(args);
             }
             else
-            {                
-                param = args[0].Trim();
-                if (args.Length > 1)
+            {
+                // Support old style
+                string param;
+                string value = null;                
+                if (args[0].Contains("="))
+                {
+                    string[] arg1 = args[0].Split(new[] { '=' }, 2);
+                    param = arg1[0].Trim();
+                    value = arg1[1].Trim();
+                }
+                else
+                {
+                    param = args[0].Trim();
+                    if (args.Length > 1)
+                    {
+                        value = args[1].Trim();
+                    }
+                }
+                HandleArg(param, value);
+            }
+
+        }
+
+        private void ParseArgs(string[] args)
+        {            
+            Parser.Default.ParseArguments<RunOptions, GridOptions>(args)
+                   .WithParsed<RunOptions>(runOptions =>
+                   {
+                       // mCLIHelper.CLIType = eCLIType.Arguments;                       
+                       mCLIHandler = new CLIArgs();
+                       ExecRunOptions(runOptions);
+                       // Console.WriteLine($"Solution: {runOptions.Solution}");
+                       // Console.WriteLine($"runset: {runOptions.Runset}");
+                       // opt = runOptions;
+                   }).WithParsed<GridOptions>(gridOptions =>
+                   {
+                       StartGrid(gridOptions);                                
+                   })
+                   .WithNotParsed(errs =>
+                   {
+                       HandleCLIParseError(errs);
+                   });
+        }
+
+        private void StartGrid(GridOptions gridOptions)
+        {
+            Console.WriteLine("Starting Ginger Grid at port: " + gridOptions.Port);
+            GingerGrid gingerGrid = new GingerGrid(gridOptions.Port);   
+            gingerGrid.Start();
+        }
+
+        private void ExecRunOptions(RunOptions runOptions)
+        {
+            Console.WriteLine($"Solution: {runOptions.Solution}");
+            Console.WriteLine($"Runset: {runOptions.Runset}");
+
+            // Dup Code!!!!!!!!!!!!!!
+            Reporter.ToLog(eLogLevel.DEBUG, "Loading Configurations...");
+            // mCLIHandler.LoadContent(configurations, mCLIHelper, WorkSpace.Instance.RunsetExecutor);
+
+            mCLIHelper.Solution = runOptions.Solution;
+            mCLIHelper.Runset = runOptions.Runset;
+            mCLIHelper.Env = runOptions.Environment;
+
+
```

Comment:
if same method will be used for all CLI options so we should not run the LoadSolution/LoadRunset/PrepareRunsetForExecution for script option

Reply:
Correct we will need to unified